js/gulp/package-task.js

```diff
@@ -54,18 +57,17 @@ const createMainPackageJson = (target, format) => (orig) => ({
             node: {
                 import: `./${mainExport}.node.mjs`,
                 require: `./${mainExport}.node.js`,
+                types: `./${mainExport}.node.d.ts`,
```

Comment:
The TypeScript docs say that the `types` property should come first: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing. Not 100% sure the reasoning since the docs don't really explain it but I think it can mess with the resolution priorities somehow.

Reply:
Weird that the order in an object matters but ok, changed.